sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -526,19 +542,26 @@ def refresh(self, **kwargs) -> None:
                         Failed to refresh configuration settings from Azure App Configuration.
                         """
         exception: Exception = RuntimeError(error_message)
+        is_failover_request = False
         try:
             self._replica_client_manager.refresh_clients()
-            active_clients = self._replica_client_manager.get_active_clients()
-
-            headers = _get_headers(
-                kwargs.pop("headers", {}),
-                "Watch",
-                self._replica_client_manager.get_client_count() - 1,
-                self._feature_flag_enabled,
-                self._feature_filter_usage,
-                self._uses_key_vault,
-            )
-            for client in active_clients:
+            self._replica_client_manager.find_active_clients()
+
+            while self._replica_client_manager.has_next_client():
+                headers = _get_headers(
+                    kwargs.pop("headers", {}),
+                    "Watch",
+                    self._replica_client_manager.get_client_count() - 1,
```

Comment:
Does client count change with every failover request?

Reply:
No but we create the header for every request from scratch at the moment. Unless we want to parse a string to change the few things we would need to then we just generate it every time.

Reply:
Yes, but we dont need to call `get_client_count()` with every iteration, right?